packages/aws-rfdk/lib/core/lib/exporting-log-group.ts

```diff
@@ -111,10 +113,24 @@ export class ExportingLogGroup extends Construct {
       threshold: 1,
     });
 
+    // Define log retention retry options to reduce the risk of the rate exceed error
+    // as the default create log group TPS is only 5. Make sure to set the timeout of log retention function
+    // to be greater than total retry time. That's because if the function that is used for a custom resource
+    // doesn't exit properly, it'd end up in retries and may take cloud formation an hour to realize that
+    // the custom resource failed.
     const logRetention = new LogRetention(this, 'LogRetention', {
       logGroupName: props.logGroupName,
       retention: retentionInDays,
+      logRetentionRetryOptions: {
+        base: Duration.millis(200),
+        maxRetries: 7,
+      },
     });
+    // referenced from cdk code: https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/aws-logs/lib/log-retention.ts#L123
+    const logRetentionFunctionConstructId = 'LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8a';
+    const logRetentionFunction = Stack.of(this).node.tryFindChild(logRetentionFunctionConstructId)!;
+    const cfnFunction = logRetentionFunction.node.defaultChild as CfnFunction;
```

Comment:
Just to confirm that this change was made in response to my previous suggestion where we'd throw an exception if we couldn't find the lambda function.

The belief here is that if the node without the specified construct ID does not exist, then `logRetentionFunction` will be `undefined` and line `132` will throw a:

```txt
Uncaught TypeError: Cannot read property 'node' of undefined
```

There may be an opportunity to make the error more developer-friendly since we have all the context here. How about this:

```suggestion
    const logRetentionFunction = Stack.of(this).node.tryFindChild(logRetentionFunctionConstructId);
    if (!logRetentionFunction) {
      throw new Error(`Could not find log retention Lambda Function with construct ID ${logRetentionFunctionConstructId} in construct tree`);
    }
    const cfnFunction = logRetentionFunction.node.defaultChild as CfnFunction;
```

Same feedback applies for `LogGroupFactory` below.

Reply:
The reason I didn't explicitly throw new error is that it will cause insufficient branch coverage and fail the build, and it's not easy to test that error branch as the log retention function is defined in cdk library. 

I tried using `findChild` function instead and it throws more explicit error. The code will be `const logRetentionFunction = Stack.of(this).node.findChild(logRetentionFunctionConstructId);` and error message will be like `Error: No child with id: 'LogRetentionaae0aa3c5b4d4f87b02d85b201efdd8'`. I think this error message is a little better than undefined message above. What do you think if we switch to this?

Reply:
Yeah, this combined with the stack trace will be sufficient context for the developer use-case I'd envisioned. Let's go with it.